src/reducers/vms.js

```diff
@@ -91,6 +91,8 @@ function vms (state, action) {
       return state.delete('selected')
     case 'VM_ACTION_IN_PROGRESS':
       return state.setIn(['vms', action.payload.vmId, 'actionInProgress', action.payload.name], action.payload.started)
+    case 'SET_VM_CONSOLES':
+      return state.setIn(['vms', action.payload.vmId, 'consoles'], action.payload.consoles) // JS object, not immutable
```

Comment:
are you really sure it is good to mix mutable and immutable objects like this? I mean if it is part of the store and it is not just the top-level namespacing object `{ vms, config, ... }` for reducers I think we should rather go with full immutable...